contracts/integrations/test/coordinator/client_test.ts

```diff
@@ -0,0 +1,634 @@
+                        synchronize: defaultOrmConfig.synchronize,
+                    },
+                );
+                app.listen(port, () => {
+                    logUtils.log(`Coordinator SERVER API (HTTP) listening on port ${port}!`);
+                });
+                return app;
+            }),
+        );
+
+        // register coordinator servers
+        [
+            [feeRecipientAddressOne, coordinatorPort],
+            [feeRecipientAddressTwo, coordinatorPort],
+            [feeRecipientAddressThree, anotherCoordinatorPort],
+        ].forEach(async ([address, port]) => {
+            await coordinatorRegistry
+                .setCoordinatorEndpoint(`${coordinatorEndpoint}${port}`)
+                .awaitTransactionSuccessAsync(
+                    { from: address },
+                    { pollingIntervalMs: constants.AWAIT_TRANSACTION_MINED_MS },
+                );
+        });
+    });
+    beforeEach(async () => {
+        signedOrder = await orderFactory.newSignedOrderAsync();
+        anotherSignedOrder = await orderFactory.newSignedOrderAsync();
+        signedOrderWithDifferentFeeRecipient = await orderFactory.newSignedOrderAsync({
+            feeRecipientAddress: feeRecipientAddressTwo,
+        });
+        signedOrderWithDifferentCoordinatorOperator = await orderFactory.newSignedOrderAsync({
+            feeRecipientAddress: feeRecipientAddressThree,
+        });
+        makerToken.setBalance(makerAddress, constants.INITIAL_ERC20_BALANCE);
+        takerToken.setBalance(takerAddress, constants.INITIAL_ERC20_BALANCE);
+    });
+    describe('test setup', () => {
+        it('should have coordinator registry which returns an endpoint', async () => {
+            const setCoordinatorEndpoint = await coordinatorRegistry
+                .getCoordinatorEndpoint(feeRecipientAddressOne)
+                .callAsync();
+            const anotherSetCoordinatorEndpoint = await coordinatorRegistry
+                .getCoordinatorEndpoint(feeRecipientAddressThree)
+                .callAsync();
+            expect(setCoordinatorEndpoint).to.equal(`${coordinatorEndpoint}${coordinatorPort}`);
+            expect(anotherSetCoordinatorEndpoint).to.equal(`${coordinatorEndpoint}${anotherCoordinatorPort}`);
+        });
+        it('should have coordinator server endpoints which respond to pings', async () => {
+            let result = await fetchAsync(`${coordinatorEndpoint}${coordinatorPort}/v2/ping`);
+            expect(result.status).to.equal(200);
+            expect(await result.text()).to.equal('pong');
+
+            result = await fetchAsync(`${coordinatorEndpoint}${anotherCoordinatorPort}/v2/ping`);
+            expect(result.status).to.equal(200);
+            expect(await result.text()).to.equal('pong');
+        });
+    });
+    // fill handling is the same for all fill methods so we can test them all through the fillOrder and batchFillOrders interfaces
+    describe('#fillOrderAsync', () => {
```

Comment:
I think it would be a worthwhile sanity check to test all the functions exposed by the client. It will also future-proof the interface.